### PR TITLE
[TD/RA] Restored save/load UI, fixed some crashes

### PR DESCRIPTION
**Pull request by wdigger**
Restoration and fixes only!

### CODE REVIEW

**Comment by hifi** [comment]
If this single commit contains crash fixes unrelated to the save/load UI, could you separate them into different commits? They can be part of the same PR but it would be better to have multiple commits that fix a single thing with a description.

**Review by hifi — APPROVED** [review]
I'm okay with this, saved and loaded just fine in TD. Not so easy to test in RA as it doesn't play yet.

@OmniBlade?

File: common/file.cpp [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    virtual const char* GetName() const;
+    virtual unsigned long GetTime() const;
+
+    virtual bool FindFirst(const char* fname);
+    virtual bool FindNext();
+    virtual void Close();
+
+private:
+    HANDLE rc;
```
Comment (OmniBlade) [inline common/file.cpp]: Can we have more descriptive member names in keeping with the general style of the original code, so something like `FindHandle` and `FindData` for these members?

**Review by OmniBlade — COMMENTED** [review]
Only suggestion I would have is to make the member names more descriptive so its easier for the next person to come along and read the code to folllow it.

File: common/file.cpp (hunk 1 above)
Comment (wdigger) [reply common/file.cpp]: Done

**Comment by hifi** [comment]
We will merge this once its rebased and @OmniBlade gives his approval.

**Review by OmniBlade — APPROVED** [review]

**Pull request merged**